lib/modules/manager/types.ts

```diff
@@ -200,7 +202,7 @@ export interface UpdateArtifactsResult {
 
 export interface UpdateArtifact<T = Record<string, unknown>> {
   packageFileName: string;
-  updatedDeps: PackageDependency<T>[];
+  updatedDeps: Upgrade<T>[];
```

Comment:
Without this change, I'm unable to access `newDigest` field 🤷

Reply:
This interface is already used by a lot of other manager implementations of artifacts updating so we should think about the implications.

Reply:
All I wanted is the field that is already present in this object, though `PackageDependency` interface prevents its usage. We receive `Upgrade<T>` in `updateDependency`, I see no reason why we can't have it in `updateArtifacts` as well.

Reply:
I think this Pr might need closing anyway? Digest needs to be updated the non artifacts way except when tags are immutable

Reply:
Tags are not guaranteed to be immutable, at least for GitHub, Docker and Go, isn't it?